### PR TITLE
PACT-611 Add Separated Material list to the Update Record Page in the Web Application

### PR DESCRIPTION
**Pull request by felixgb**
Example of separated material:
Screenshot 2023-01-10 at 15 52 55

Example with no separated material:
Screenshot 2023-01-10 at 15 53 08

### CODE REVIEW

File: test/support/CustomMatchers.scala [hunk 1]
```diff
@@ -188,7 +188,21 @@ object CustomMatchers {
...
+      singleValueMatcher(
+        label = "a single list item with the text None",
+        expectedValue = "None",
+        actualValue = document.select("#separated-material > li").text()
+      )
+    else
+      singleValueMatcher(
+        label = "a list of separated material",
+        expectedValue = separatedMaterials.toSeq,
+        actualValue = getSeperatedMaterialItems(document).toSeq
```
Comment (tna-erasmos) [inline test/support/CustomMatchers.scala:205]: You can remove `toSeq`.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html [hunk 2]
```diff
@@ -48,7 +48,7 @@
     ))
 }
 
-@noRelatedMaterial = {
+@noListItem = {
   <li>None</li>
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html:52]: Should be a resolved message.

File: app/uk/gov/nationalarchives/omega/editorial/models/package.scala [hunk 3]
```diff
@@ -57,6 +57,20 @@ package object editSetRecords {
         |      "description" : "Bedlington Colliery, Newcastle Upon Tyne. Photograph depicting: view of pithead baths."
         |    }
         |  ],
+        |  "separatedMaterial" : [
```
Comment (tna-erasmos) [inline app/uk/gov/nationalarchives/omega/editorial/models/package.scala:60]: I recommend that we have a variety of item types (e.g. some with a link, some with a description) whether in one record or (preferably) across several records; if only to better demonstrate that variety.

**Review by tna-erasmos — COMMENTED** [review]
Minor changes suggested.

File: test/support/CustomMatchers.scala (hunk 1 above)
Comment (tna-erasmos) [reply test/support/CustomMatchers.scala]: Confirmed.

File: app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html (hunk 2 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/views/editSetRecordEdit.scala.html]: Confirmed.

File: app/uk/gov/nationalarchives/omega/editorial/models/package.scala (hunk 3 above)
Comment (tna-erasmos) [reply app/uk/gov/nationalarchives/omega/editorial/models/package.scala:60]: Confirmed.

**Review by tna-erasmos — APPROVED** [review]
LGTM

File: app/uk/gov/nationalarchives/omega/editorial/models/package.scala (hunk 3 above)
Comment (felixgb) [reply app/uk/gov/nationalarchives/omega/editorial/models/package.scala:60]: I've added different separated material types to records 10, 11 and 12

**Review by rwalpole — APPROVED** [review]

**Comment by userjd** [comment]
@rwalpole could you confirm if it would be useful to have tests for a record with the Related and Separated material with the scenarios as follows: 1) both present 2) both not present at all 3) only Related material 4) only Separated material.  Primarily all the tests for form validation /POST  Presently, the existence of Related and Separated material  scenario (1) is checked in the `/GET /edit-set/{id}/record/{recordId}/edit`  scenario `all data is valid`  . And scenario (4) being is checked in the `'background' field is blank` 
` separatedMaterial = Seq.empty`

To keep it consistent these could be added to the /GET similar to this scenario `all data is valid`, so another for a record with Separated material but no Related material (and vice-versa) with all the other fields as required.

File: test/controllers/EditSetControllerSpec.scala [hunk 4]
```diff
@@ -948,6 +948,20 @@ class EditSetControllerSpec extends BaseSpec {
...
+              ),
+              ExpectedSeparatedMaterial(
+                linkHref = Some("#;"),
+                linkText = Some("COAL 80/80/6")
+              ),
+              ExpectedSeparatedMaterial(
+                linkHref = Some("#;"),
+                linkText = Some("COAL 80/80/7")
+              )
+            ),
```
Comment (userjd) [inline test/controllers/EditSetControllerSpec.scala:964]: `when all data is valid` scenario to check for existence of Related and Separated. Suggestion to include test when only Separated or Related exists

**Review by userjd — APPROVED** [review]

**Pull request merged**